### PR TITLE
First version of a linear function approximation model

### PR DESCRIPTION
**Pull request by tkrieger**
This PR adds a first model version for the auto_bomber agent.
Specifically:
- linear function approximation using gradient descent for Q-Learning
- monte carlo method for *episode* based training
- epsilon greedy policy for act in case of training

--> feature extraction simply creates a random array & rewards are only a first idea

### CODE REVIEW

**Review by GucciGui — DISMISSED** [review]

File: agent_code/auto_bomber/model.py [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+import agent_code.auto_bomber.auto_bomber_config as config
+from agent_code.auto_bomber.transitions import Transitions
+
+
+class LinearAutoBomberModel:
+    def __init__(self, feature_extractor):
+        self.weights = None
+        self.feature_extractor = feature_extractor
+
+        if os.path.isfile(config.MODEL_PATH):
```
Comment (PrimeF) [inline agent_code/auto_bomber/model.py]: Best practice in Python 3 is currently to use the object-oriented approach for handling paths, i.e. pathlib.Path.

See https://docs.python.org/3/library/pathlib.html#correspondence-to-tools-in-the-os-module for conversion table from os module to path lib module.

In our case I guess it is optional to migrate to the pathlib approach, just a suggestion from my side.

File: agent_code/auto_bomber/test_transitions.py [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+
+        x_0_all, y_0_all = numpy_trans.get_features_and_value_estimates(config.ACTIONS[0])
+        self.assertEqual((1, 10), x_0_all.shape)
+        self.assertEqual((1,), y_0_all.shape)
+
+
+
+
+
+
```
Comment (PrimeF) [inline agent_code/auto_bomber/test_transitions.py]: Blank lines.

File: agent_code/auto_bomber/test_model.py [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+            np.testing.assert_array_equal(model.weights[3, :], np.ones(3,))
+            np.testing.assert_array_equal(model.weights[4, :], np.ones(3,))
+            np.testing.assert_array_equal(model.weights[5, :], np.ones(3,))
+
+
+
+
+
+
+
```
Comment (PrimeF) [inline agent_code/auto_bomber/test_model.py]: Blank lines.

**Review by PrimeF — CHANGES_REQUESTED** [review]
Just a couple of Python syntactical suggestions from my side.

Otherwise LGTM.

File: agent_code/auto_bomber/model.py (hunk 1 above)
Comment (tkrieger) [reply agent_code/auto_bomber/model.py]: Oh I see, this maybe even simplifies mocking, I will change this.
Seems that the agent template code where I copied the stuff from is also a bit dated

File: agent_code/auto_bomber/test_transitions.py (hunk 2 above)
Comment (tkrieger) [reply agent_code/auto_bomber/test_transitions.py]: Nap good catch, I guess I will correct my auto formatter :D

**Review by PrimeF — APPROVED** [review]
LGTM

**Pull request merged**